pkg/kubelet/apis/podresources/server_test.go

```diff
@@ -0,0 +1,153 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package podresources
+
+import (
+	"context"
+	"testing"
+
+	"github.com/stretchr/testify/mock"
+
+	"k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/kubernetes/pkg/kubelet/apis/podresources/v1alpha1"
+)
+
+type mockProvider struct {
+	mock.Mock
+}
+
+func (m *mockProvider) GetPods() []*v1.Pod {
+	args := m.Called()
+	return args.Get(0).([]*v1.Pod)
+}
+
+func (m *mockProvider) GetDevices(podUID, containerName string) []*v1alpha1.ContainerDevices {
+	args := m.Called(podUID, containerName)
+	return args.Get(0).([]*v1alpha1.ContainerDevices)
+}
+
+func TestListPodResources(t *testing.T) {
+	podName := "pod-name"
+	podNamespace := "pod-namespace"
+	podUID := types.UID("pod-uid")
+	containerName := "container-name"
+
+	devs := []*v1alpha1.ContainerDevices{
+		{
+			ResourceName: "resource",
+			DeviceIds:    []string{"dev0", "dev1"},
+		},
+	}
+
+	for _, tc := range []struct {
+		desc             string
+		pods             []*v1.Pod
+		devices          []*v1alpha1.ContainerDevices
+		expectedResponse *v1alpha1.ListPodResourcesResponse
+	}{
+		{
+			desc:             "no pods",
+			pods:             []*v1.Pod{},
+			devices:          []*v1alpha1.ContainerDevices{},
+			expectedResponse: &v1alpha1.ListPodResourcesResponse{},
+		},
+		{
+			desc: "pod without devices",
+			pods: []*v1.Pod{
+				{
```

Comment:
Do you think a helper would make more sense here?

Reply:
We only make 2 pods, so I don't think its worth it.  if/when we add to these, it may make more sense to add a helper.